File: python/剑指offer/面试题31. 栈的压入、弹出序列.py
from typing import List
class Solution:
    def validateStackSequences(self, pushed: List[int], popped: List[int]) -> bool:
        stack, i = [],0
        for num in pushed:
            stack.append(num)
            while stack and stack[-1] == popped[i]:
                stack.pop()
                i+=1
        return not stack


s = Solution()
pushed = [2,1,0]
popped = [1,2,0]
print(s.validateStackSequences(pushed,popped))

File: go/剑指offer/面试题31. 栈的压入、弹出序列.go
package main

func validateStackSequences(pushed []int, popped []int) bool {
	var stack []int
	var i int
	for _, num := range pushed {
		stack = append(stack, num)
		for len(stack) != 0 && stack[len(stack)-1] == popped[i] {
			stack = stack[:len(stack)-1]
			i += 1
		}
	}
	return len(stack) == 0
}
